contracts/craft/Salvage.sol

```diff
@@ -64,61 +72,91 @@ contract Salvage is ISalvage, CraftBase {
 
         for (uint256 i = 0; i < _assets.length; ++i) {
-            salvage(_assets[i], _amounts[i]);
+            _salvage(_assets[i], _amounts[i]);
         }
         
-        // Todo: update this to include the salvage rewards
         emit AssetSalvagedBatch(_msgSender(), _assets, _amounts);
     }
 
-    function getSalvageRewards(LibCraft.AssetData calldata _asset) external view override returns(LibCraft.SalvageReward[] memory rewards) {
-        rewards = salvageableAssets[_getId(_asset.content, _asset.tokenId)].rewards;
+    /**
+    * @dev query the list of salvage outputs available (and lootbox credits, if any)
+    * @param _asset the asset to query
+    */
+    function getSalvageOutputs(LibCraft.AssetData calldata _asset) external view override returns(LibCraft.SalvageOutput[] memory outputAssets, LibLootbox.LootboxCreditReward memory outputLootboxCredits) {
+        outputAssets = salvageableAssets[_asset.hashAssetId()].outputs;
+        outputLootboxCredits = salvageableAssets[_asset.hashAssetId()].lootboxCredits;
     }
 
     /**************** Internal Functions ****************/
-    function _getId(address _content, uint256 _tokenId) internal pure returns(uint256) {
-        return uint256(keccak256(abi.encodePacked(_content, _tokenId)));
-    }
 
-    function _validateInput(LibCraft.AssetData memory _asset, uint256 _amount) internal view {
-        // 1. Check if the asset is salvageable
-        require(salvageableAssets[_getId(_asset.content, _asset.tokenId)].rewards.length > 0, "Item not salvageable");
-        // 2. Check amount > 0
-        require(_amount > 0, "Invalid amount");
+    function _salvage(LibCraft.AssetData memory _asset, uint256 _amount) internal {
+        require(_amount > 0, "Error: Invalid amount");
+
+        uint256 id = _asset.hashAssetId();
+        
+        LibCraft.SalvageOutput[] memory outputAssets;
+        uint256[] memory amounts;
+
+        if (LibCraft.SalvageType(salvageableAssets[id].salvageType) == LibCraft.SalvageType.Guarantee) {
+            outputAssets = salvageableAssets[id].outputs;
+            amounts = new uint256[](salvageableAssets[id].outputs.length);
+            for (uint256 i = 0; i < salvageableAssets[id].outputs.length; ++i) {
+                amounts[i] = salvageableAssets[id].outputs[i].amount.mul(_amount);
+            }
+        } else if (LibCraft.SalvageType(salvageableAssets[id].salvageType) == LibCraft.SalvageType.Random) {
+            outputAssets = salvageableAssets[id].outputs;
+            amounts = new uint256[](salvageableAssets[id].outputs.length);
+            
+            // Get the number of passing rolls
+            for (uint256 i = 0; i < _amount; ++i) {
+                for (uint256 j = 0; j < salvageableAssets[id].outputs.length; ++j) {
+                    seed = LibCraft.random(_msgSender(), seed);
+                    if (seed.mod(1e6) > (1e6 - salvageableAssets[id].outputs[j].probability)) {
```

Comment:
Change to "if (seed.mod(1e6) < (salvageableAssets[id].outputs[j].probability)) {" to match how it's done in the other contracts.